### ISSUE TITLE
XL Task Error: Run failed, Exception: Object reference not set to instance of an object

### ISSUE THREAD

**Issue by mriffle** [smith-chem-wisc/MetaMorpheus#1100]
Greetings, I am attempting to run a XL task to test the latest output from MetaMorpheus. I have asked the program to output a pepXML file and output cross-links, loop-links, and mono-links. Here is my toml file:

```
TaskType = "XLSearch"

[XlSearchParameters]
FragmentationType = "MS2_HCD"
DecoyType = "Reverse"
DoLocalizationAnalysis = false
CrosslinkerType = "DSS"
CrosslinkSearchTop = false
CrosslinkSearchTopNum = 300
CrosslinkSearchWithAllBeta = false
UdXLkerCleavable = false
UdXLkerResidue = 75
XlPrecusorMsTl = "±10.0000 PPM"
XlBetaPrecusorMsTl = "±10.0000 PPM"
DisposeOfFileWhenDone = false
XlOutPercolator = false
XlOutAll = true
XlOutPepXML = true
XlOutCrosslink = true
XlQuench_H2O = true
XlQuench_Tris = true
XlQuench_NH2 = false
XlCharge_2_3 = true
XlCharge_2_3_PrimeFragment = false

[CommonParameters]
MaxParallelFilesToAnalyze = 1
MaxThreadsToUsePerFile = 1
LocalizeAll = true
ListOfModsFixed = "Common Fixed\tCarbamidomethyl of C\t\tCommon Fixed\tCarbamidomethyl of U"
ListOfModsVariable = "Common Variable\tOxidation of M"
DoPrecursorDeconvolution = true
UseProvidedPrecursorInfo = true
DeconvolutionIntensityRatio = 3.0
DeconvolutionMaxAssumedChargeState = 12
DeconvolutionMassTolerance = "±4.0000 PPM"
TotalPartitions = 1
BIons = true
YIons = true
ZdotIons = false
CIons = false
ProductMassTolerance = "±20.0000 PPM"
PrecursorMassTolerance = "±5.0000 PPM"
ConserveMemory = true
ScoreCutoff = 5.0
ReportAllAmbiguity = true
TopNpeaks = 200
MinRatio = 0.01
TrimMs1Peaks = false
TrimMsMsPeaks = true
TaskDescriptor = "XLSearchTask"
CalculateEValue = false

[CommonParameters.DigestionParams]
MaxMissedCleavages = 2
InitiatorMethionineBehavior = "Variable"
MinPeptideLength = 5
MaxModificationIsoforms = 1024
MaxModsForPeptide = 2
Protease = "trypsin"
SemiProteaseDigestion = false
TerminusTypeSemiProtease = "None"
```

The program runs and generates a allPsms.mytsv, xl_inter_fdr.mytsv, xl_intra_fdr.mytsv, and a results.txt file. The allPsms.mytsv file looks like it contains results. However, it then crashes with the following contents of the results.txt file. There is no pepXML file.

```
MetaMorpheus: version 0.0.276.0
Data files were processed on a innotek GmbH VirtualBox computer using Windows version 6.2.9200.0 with a 64-Bit processor and 1 cores operating at 2.395GHz and 0GB installed RAM.
.NET Framework Version: 4.7 or laterThermo MSFileReader Fileio DLL		Version: 3.1
Thermo MSFileReader Fregistry DLL		Version: 3.1
Thermo MSFileReader Rawfile Com Interface		Version: 3.1.5.0

e: System.NullReferenceException: Object reference not set to an instance of an object.
   at TaskLayer.XLSearchTask.<>c.<RunSpecific>b__7_15(PsmCross p) in C:\projects\metamorpheus\TaskLayer\XLSearchTask\XLSearchTask.cs:line 318
   at System.Linq.Enumerable.WhereListIterator`1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.OrderedEnumerable`1.<GetEnumerator>d__1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at TaskLayer.XLSearchTask.RunSpecific(String OutputFolder, List`1 dbFilenameList, List`1 currentRawFileList, String taskId, FileSpecificSettings[] fileSettingsList) in C:\projects\metamorpheus\TaskLayer\XLSearchTask\XLSearchTask.cs:line 318
   at TaskLayer.MetaMorpheusTask.RunTask(String output_folder, List`1 currentProteinDbFilenameList, List`1 currentRawDataFilepathList, String displayName) in C:\projects\metamorpheus\TaskLayer\MetaMorpheusTask.cs:line 231e.Message: Object reference not set to an instance of an object.e.InnerException: e.Source: TaskLayere.StackTrace:    at TaskLayer.XLSearchTask.<>c.<RunSpecific>b__7_15(PsmCross p) in C:\projects\metamorpheus\TaskLayer\XLSearchTask\XLSearchTask.cs:line 318
   at System.Linq.Enumerable.WhereListIterator`1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.OrderedEnumerable`1.<GetEnumerator>d__1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at TaskLayer.XLSearchTask.RunSpecific(String OutputFolder, List`1 dbFilenameList, List`1 currentRawFileList, String taskId, FileSpecificSettings[] fileSettingsList) in C:\projects\metamorpheus\TaskLayer\XLSearchTask\XLSearchTask.cs:line 318
   at TaskLayer.MetaMorpheusTask.RunTask(String output_folder, List`1 currentProteinDbFilenameList, List`1 currentRawDataFilepathList, String displayName) in C:\projects\metamorpheus\TaskLayer\MetaMorpheusTask.cs:line 231e.TargetSite: Boolean <RunSpecific>b__7_15(EngineLayer.CrosslinkSearch.PsmCross)
```

Appreciate any advice.

**Comment by lonelu** [comment]
Hi. I suppose you are using a old toml file generated before(before version 273). We recently updated toml management code. And the old toml will crash the software like that.
Please generate a new toml file and try again. Sorry for the inconvenience.

**Comment by lonelu** [comment]
Hi. I suppose you are using a old toml file generated before(before version 273). We recently updated toml management code. And the old toml will crash the software like that.
Please generate a new toml file and try again. Sorry for the inconvenience. Please let me know if it still not work.

**Comment by mriffle** [comment]
Hi, thanks for the quick response. This toml file was actually generated by the current version of the software (276), right as I was attempting to run it.

**Comment by mriffle** [comment]
I did install 276 over an older version of the software, if that is helpful.

**Comment by lonelu** [comment]
That's very helpful.  I will figure out this soon.

**Comment by rmillikin** [comment]
Thanks for the report, should be fixed in #1101 . I'll likely be making a new release tomorrow which will include this change.

**Comment by rmillikin** [comment]
Slight delay in the release, sorry about that. It's out now; this issue should be fixed in v0.0.277.
